src/natcap/invest/recreation/recmodel_client.py

```diff
@@ -1502,6 +1505,33 @@ def error_handler(err_level, err_no, err_msg):
             "projection of the base vector")
 
 
+def _validate_predictor_types(table_path):
+    """Validate the type values in a predictor table.
+
+    Args:
+        table_path (string): path to a csv table that has at least
+            the field 'type' 
+
+    Returns:
+        None
+
+    Raises:
+        ValueError if any value in the ``type`` column does not match a valid
+        type, ignoring leading/trailing whitespace.
+    """
+    df = utils.read_csv_to_dataframe(table_path, to_lower=True)
+    type_list = df['type'].tolist()
```

Comment:
I think what you have here is just fine, but I wonder if using `sets` would be a bit less code and cleaner?
```python
type_set = set(df['type'].str.strip().tolist())
valid_types = set([ ... ])
invalid_types = type_set.difference(valid_types)
if invalid_types:
    raise ValueError(...)
```

Sorry if this feels like over designing, I've just been loving `sets` for list comparisons recently.

Reply:
good idea! Sets are very cool.